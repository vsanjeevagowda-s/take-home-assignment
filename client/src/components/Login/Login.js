import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import {
  Box,
  Button,
  FormControl,
  TextField,
  FormLabel,
  FormGroup,
  Typography,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import Alert from "@material-ui/lab/Alert";
import { login } from "../../services/user";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
  },
  button: {
    backgroundColor: "#005A3C",
    color: "white",
  },
  link: {
    color: "#005A3C",
    marginTop: "1em",
  },
  formWrapper: {
    height: "100vh",
  },
}));

const Login = () => {
  const classes = useStyles();
  const history = useHistory();

  const [email, setEmail] = useState("user@gmail.com");
  const [password, setPassword] = useState("StrongPassword123");
  const [errorMessage, setErrorMessage] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();

    login({ email, password })
      .then((resp) => {
        console.log('===>Inside login then<===', resp)
        console.log('===>history<===', history)
        history.push("/dashboard");
      })
      .catch((error) => {
        console.log('===>Inside login error<===', error);
        setErrorMessage(error.message);
        setTimeout(() => setErrorMessage(""), 2000);
      });
  };

  return (
    <Box
      display="flex"
      justifyContent="center"
      alignItems="center"
      bgcolor="inherit"
      className={classes.formWrapper}
    >
      <Box p={4} bgcolor="#FFFFFF" width="40%" borderRadius={16}>
        <Box p={2} display="flex" justifyContent="center">
          <Typography variant="h5">Sign In</Typography>
        </Box>

        {errorMessage && (
          <Box display="flex" justifyContent="center">
            <Alert data-testid="error-message" severity="error">{errorMessage}</Alert>
          </Box>
        )}
        <form
          className={classes.root}
          noValidate
          autoComplete="off"
          onSubmit={handleSubmit}
        >
          <FormGroup>
            <FormControl>
              <FormLabel>
                <Box p={1}>Email</Box>
              </FormLabel>
              <TextField
                inputProps={{ "data-testid": "login-email" }}
                type="text"
                variant="outlined"
                onChange={(event) => setEmail(event.target.value)}
                value={email}
              />
            </FormControl>
          </FormGroup>
          <FormGroup>
            <FormControl>
              <FormLabel>
                <Box p={1}>Password</Box>
              </FormLabel>
              <TextField
                inputProps={{ "data-testid": "login-password" }}
                name="password"
                type="password"
                variant="outlined"
                onChange={(event) => setPassword(event.target.value)}
                value={password}
              />
            </FormControl>
          </FormGroup>
          <Box mt={2}>
            <Button data-testid="login-button" className={classes.button} type="submit" fullWidth>
              Login
            </Button>
          </Box>
        </form>
        <Box display="flex" justifyContent="center">
          <Link to="/register" className={classes.link}>
            Create an account
          </Link>
        </Box>
      </Box>
    </Box>
  );
};

export default Login;
