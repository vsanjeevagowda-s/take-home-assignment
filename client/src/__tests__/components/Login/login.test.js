import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Login from "../../../components/Login/Login";

describe('login', () => {
  test("renders login component", () => {
    const { getByText } = render(<Login />);
    const linkElement = getByText(/Sign In/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("renders email field", () => {
    const { getByTestId } = render(<Login />);
    const ele = getByTestId('login-email')
    fireEvent.change(ele, { target: { value: 'a@gmail.com' } })
    expect(ele.value).toBe('a@gmail.com')
  });

  test("renders password field", () => {
    const { getByTestId } = render(<Login />);
    const ele = getByTestId('login-password')
    fireEvent.change(ele, { target: { value: '123' } })
    expect(ele.value).toBe('123')
  });
});
