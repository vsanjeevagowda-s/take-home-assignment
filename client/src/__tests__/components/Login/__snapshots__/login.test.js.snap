// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register should login with valid email / password 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-50 makeStyles-formWrapper-49"
  >
    <div
      class="MuiBox-root MuiBox-root-51"
    >
      <div
        class="MuiBox-root MuiBox-root-52"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          Sign Up
        </h5>
      </div>
      <form
        autocomplete="off"
        class="makeStyles-root-46"
        novalidate=""
      >
        <div
          class="MuiFormGroup-root"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root"
            >
              <div
                class="MuiBox-root MuiBox-root-53"
              >
                Email
              </div>
            </label>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="register-email"
                  type="text"
                  value="user@gmail.com"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-54 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-55"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiFormGroup-root"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root"
            >
              <div
                class="MuiBox-root MuiBox-root-58"
              >
                Password
              </div>
            </label>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="register-password"
                  name="password"
                  type="password"
                  value="password"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-54 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-55"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-59"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-47 MuiButton-fullWidth"
            data-testid="register-button"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Create an account
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
      <div
        class="MuiBox-root MuiBox-root-60"
      >
        <a
          class="makeStyles-link-48"
          href="#/"
        >
          Have an account already? Login here
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
